src/mono/wasm/templates/templates/console/app-support.cjs

```diff
@@ -0,0 +1,427 @@
+                return;
+            }
+
+            Module.FS.chdir(processedArguments.working_dir);
+
+            if (processedArguments.runtime_args.length > 0)
+                INTERNAL.mono_wasm_set_runtime_options(processedArguments.runtime_args);
+
+            console.info("Initializing.....");
+            Object.assign(App, { MONO, INTERNAL, BINDING, Module, processedArguments });
+
+            try {
+                if (App.init)
+                {
+                    let ret = App.init();
+                    Promise.resolve(ret).then(function (code) { set_exit_code(code ?? 0); });
+                }
+                else
+                {
+                    console.log("WASM ERROR: no App.init defined");
+                    set_exit_code(1, "WASM ERROR: no App.init defined");
+                }
+            } catch (err) {
+                console.log(`WASM ERROR ${err}`);
+                if (is_browser && document.getElementById("out"))
+                    document.getElementById("out").innerHTML = `error: ${err}`;
+                set_exit_code(1, err);
+            }
+        },
+        onAbort: (error) => {
+            set_exit_code(1, stringify_as_error_with_stack(new Error()));
+        },
+    }))
+}).catch(function (err) {
+    set_exit_code(1, "failed to load the dotnet.js file.\n" + stringify_as_error_with_stack(err));
+});
+
+function set_exit_code(exit_code, reason) {
+    if (reason) {
+        if (reason instanceof Error)
+            console.error(stringify_as_error_with_stack(reason));
+        else if (typeof reason == "string")
+            console.error(reason);
+        else
+            console.error(JSON.stringify(reason));
+    }
+
+    if (is_browser) {
+        if (App.Module) {
+            // Notify the selenium script
+            App.Module.exit_code = exit_code;
+        }
+
+        //Tell xharness WasmBrowserTestRunner what was the exit code
+        const tests_done_elem = document.createElement("label");
+        tests_done_elem.id = "tests_done";
+        tests_done_elem.innerHTML = exit_code.toString();
+        document.body.appendChild(tests_done_elem);
+
+        if (forward_console)  {
+            const stop_when_ws_buffer_empty = () => {
+                if (consoleWebSocket.bufferedAmount == 0) {
+                    // tell xharness WasmTestMessagesProcessor we are done. 
+                    // note this sends last few bytes into the same WS
+                    console.log("WASM EXIT " + exit_code);
+                }
+                else {
+                    setTimeout(stop_when_ws_buffer_empty, 100);
+                }
+            };
+            stop_when_ws_buffer_empty();
+        } else {
+            console.log("WASM EXIT " + exit_code);
+        }
+
+    } else if (App && App.INTERNAL) {
+        App.INTERNAL.mono_wasm_exit(exit_code);
+    }
+}
+
+async function loadDotnet(file) {
```

Comment:
if the template is ES6, let's stick with `import createDotnetRuntime from './dotnet.js'`
this complex way is there to support all possible combinations.